### ISSUE TITLE
`systemd-modules-load` doesn't list `libkmod.so*` as required

### ISSUE THREAD

**Issue by Winterhuman** [systemd/systemd#35541]
### systemd version the issue has been seen with

257

### Used distribution

Arch Linux

### Linux kernel version used

6.12.4-arch1-1

### CPU architectures issue was seen on

x86_64

### Expected behaviour you didn't see

`dlopen-notes` or `readelf --dynamic` list `libkmod.so*` libraries as required, or at least recommended, since otherwise `systemd-modules-load.service` fails on most systems.

### Unexpected behaviour you saw

`libkmod.so*` isn't listed by either tools, but is required for `systemd-modules-load` to succeed.

### Steps to reproduce the problem

1. `{readelf --dynamic,dlopen-notes --sonames} /usr/lib/systemd/systemd-modules-load`
2. Observe that the `libkmod.so*` libraries aren't included in either ones results.

**Comment by Winterhuman** [comment]
That's true, but isn't it still correct to list `libkmod.so*` as a required dependency of `systemd-modules-load`? It still feels weird that `readelf --dynamic` won't report it

**Comment by yuwata** [comment]
At least, currently not supported, and packager manually sets the dependency in such case.

**Comment by Winterhuman** [comment]
Is it realistically possible to make `systemd-modules-load` succeed without `libkmod.so*`? I wouldn't think there would be any other way for it to achieve its function (unless you mean that `systemd-modules-load` is technically optional since you could build-in all your modules). What do you mean by "packager" btw?

**Comment by yuwata** [comment]
> Is it realistically possible to make `systemd-modules-load` succeed without `libkmod.so*`?

I do not think so?

> What do you mean by "packager" btw?

I meant the downstream maintainer. E.g. Fedora manually adds dependency to the package main and udev sub package.
https://src.fedoraproject.org/rpms/systemd/c/7740aea610b37fbac557cf4e73ae35bfa49b8b13

**Comment by bluca** [comment]
This is a packaging issue, the shared library dlopens kmod, and the executable depends on the shared library. Dlopen recommends must always be installed unless there are very specific reasons not to.

**Comment by Winterhuman** [comment]
Oh, so "recommended" dependencies should basically be treated as "required"? "recommended" feels like strange wording, and https://systemd.io/ELF_PACKAGE_METADATA doesn't mention how those strings should be interpreted, but, that's not a systemd issue at that point. Opening https://github.com/systemd/package-notes/issues/46 instead.

To clarify, I ran into this while trimming an initramfs down to size with my own script, not directly after I installed with a package manager (`pacman -Qi systemd` lists `kmod` as a required dependency as expected), and I ran into this issue from accidentally skipping the dlopen check on the `libshared*` library.

Closing issue.
